tensorflow/core/kernels/map_kernels.h

```diff
@@ -176,6 +174,39 @@ class TensorMapHasKey : public OpKernel {
   }
 };
 
+class TensorMapStackKeys : public OpKernel {
+ public:
+  explicit TensorMapStackKeys(OpKernelConstruction* c) : OpKernel(c) {
+    OP_REQUIRES_OK(c, c->GetAttr("key_dtype", &key_dtype_));
+  }
+  ~TensorMapStackKeys() override {}
+
+  void Compute(OpKernelContext* c) override {
+    const TensorMap* m = nullptr;
+    OP_REQUIRES_OK(c, GetInputMap(c, 0, &m));
+    
+    OP_REQUIRES(c, m->size() != 0,
+                errors::InvalidArgument("Empty map has no keys."));
+
+    auto it = m->tensors().begin();
+    size_t sz = m->tensors().size();
+    TensorShape shape = it->first.shape();
+    shape.InsertDim(0, m->tensors().size());
+    Tensor* result;
+    OP_REQUIRES_OK(c, c->allocate_output(0, shape, &result));
+    int i = 0;
+    while (it != m->tensors().end() && i < sz) {
+      OP_REQUIRES(c, it->first.dtype() == key_dtype_,
+                  errors::InvalidArgument("Key has mismatched dtype."));
```

Comment:
To the user, this means rather that the map has keys of mismatched dtypes - can you update the message accordingly?